pkg/controller/controllercmd/cmd.go

```diff
@@ -163,6 +163,7 @@ func (c *ControllerCommandConfig) StartController(stopCh <-chan struct{}) error
 		WithLeaderElection(config.LeaderElection, "", c.componentName+"-lock").
 		WithServer(config.ServingInfo, config.Authentication, config.Authorization).
 		WithFileObserver(fileobserver.ExitOnChangeReactor, observedFiles...).
+		WithEventRecorder().
```

Comment:
Calling this is optional? No nil pointer panic? This should be documented in Godoc of WithEventRecorder.